### PR TITLE
7 4 x evt gen undecay patch

### PR DESCRIPTION
**Pull request by inugent**
Adding patch for bug found by Mikhail in hepmc event record. The final patch is based on his code.

### CODE REVIEW

**Comment by bendavid** [comment]
cout is not allowed in production code.  Please use edm::LogInfo or edm::LogWarning

**Comment by bendavid** [comment]
Is this related to https://hypernews.cern.ch/HyperNews/CMS/get/relval/3331.html
?

**Comment by degano** [comment]
@lveldere please remove the std::cout so we can proceed with tests.

**Comment by lveldere** [comment]
which cout, where?


On Wed, Jan 21, 2015 at 3:14 PM, Alessandro Degano <notifications@github.com

**Comment by bendavid** [comment]
@mkirsano

**Comment by degano** [comment]
Sorry I pointed the wrong person. I mean @inugent.

**Comment by degano** [comment]
@cmsbuild please test

File: GeneratorInterface/EvtGenInterface/plugins/EvtGen/EvtGenInterface.cc [hunk 1]
```diff
@@ -428,6 +428,12 @@ HepMC::GenEvent* EvtGenInterface::decay( HepMC::GenEvent* evt ){
       addToHepMC(*p,idEvt,evt);                     // generate decay
     }
   }
+
+  // add code to ensure all particles have an end vertex and if they are undecayed with no end vertes set to status 1
+  for (HepMC::GenEvent::particle_const_iterator p= evt->particles_begin(); p != evt->particles_end(); ++p){
+    if((*p)->end_vertex() && (*p)->status() == 1)(*p)->set_status(2);
+    if((*p)->end_vertex() && (*p)->end_vertex()->particles_out_size()==0) std::cout << "EvtGenInterface::decay error: empty end vertex!" <<    std::endl;
+  }
```
Comment (davidlange6) [inline GeneratorInterface/EvtGenInterface/plugins/EvtGen/EvtGenInterface.cc]: use the message logger (or nothing) instead of cout.

**Comment by bendavid** [comment]
-1
This can be closed in favour of https://github.com/cms-sw/cmssw/pull/7355

**Comment by bendavid** [comment]
Why was this merged when it was rejected?  Logic error in cmsbuild bot?

**Comment by slava77** [comment]
It could be  just a note indicating a result of a merge in another PR subsuming this one, #7355 
contains 33c693e

**Comment by bendavid** [comment]
@slava77 I think you're right, and indeed the 7_4_X branch only shows a merge commit for https://github.com/cms-sw/cmssw/pull/7355 and not for this one, so things look as they should be (and just the message at the top of this page is slightly confusing)

**Pull request merged**